core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -985,15 +985,23 @@ public void addFailedEngineListener(Engine.FailedEngineListener failedEngineList
     }
 
     public void updateBufferSize(ByteSizeValue shardIndexingBufferSize, ByteSizeValue shardTranslogBufferSize) {
+        Engine engine = engineUnsafe();
```

Comment:
Good catch, I'll fix it to set the config even if engine is null, and add javadocs about when engineUnsafe can return null...